custom/icds_reports/tasks.py

```diff
@@ -1221,14 +1221,29 @@ def aggregate_validation_helper():
                                                                        'wer_weighed', 'wer_eligible',
                                                                        'incentive_eligible')
                                      .order_by('awc_id'))
-    awc_aggregate_rows_list = list(AggAwc.objects.values('awc_id', 'is_launched',
-                                                         'valid_visits', 'expected_visits')
+    awc_aggregate_rows_list = list(AggAwc.objects.filter(aggregation_level=5, month=agg_date)
+                                   .values('awc_id', 'is_launched', 'valid_visits', 'expected_visits')
                                    .order_by('awc_id'))
     is_launched_check_bad_data = []
     home_conduct_check_bad_data = []
     eligibility_check_bad_data = []
-    for index, awc_from_performance in enumerate(awc_performance_rows_list):
-        awc_from_aggregate = awc_aggregate_rows_list[index]
+    missed_ids_from_performance = []
+    performance_index = 0
+    aggregate_index = 0
+    while performance_index < len(awc_performance_rows_list) and aggregate_index < len(awc_aggregate_rows_list):
+        awc_from_performance = awc_performance_rows_list[performance_index]
+        awc_from_aggregate = awc_aggregate_rows_list[aggregate_index]
+        awc_id_from_performance = awc_from_performance['awc_id']
+        awc_id_from_aggregate = awc_from_aggregate['awc_id']
+        # skipping unmatched rows
+        while awc_id_from_performance != awc_id_from_aggregate:
+            if awc_id_from_performance > awc_id_from_aggregate:
+                missed_ids_from_performance.append(awc_id_from_performance)
+                aggregate_index += 1
+            else:
+                performance_index += 1
```

Comment:
This should be logged as missing as well. Just like missing rows in the performance report.